clinica/pipelines/engine.py

```diff
@@ -748,35 +772,12 @@ def copy2_add_ses(src, dst):
                 for f in listdir(bids_dir)
                 if isdir(join(bids_dir, f)) and f.startswith("sub-")
             ]
+
             cross_subj = []
             long_subj = []
-            for sub in all_subs:
-                # Use is_cross_sectional to know if the current subject needs to be
-                # labeled as cross sectional
-                is_cross_sectional = False
-                folder_list = [
-                    f
-                    for f in listdir(join(bids_dir, sub))
-                    if isdir(join(bids_dir, sub, f))
-                ]
-                for fold in folder_list:
-                    if not fold.startswith("ses-"):
-                        is_cross_sectional = True
-                if is_cross_sectional:
-                    cross_subj.append(sub)
-                else:
-                    long_subj.append(sub)
-
-            # The following code is run if cross sectional subjects have been found
-            if len(cross_subj) > 0:
-                cprint(
-                    msg=(
-                        f"{len(cross_subj)} subjects in your BIDS folder did not respect the longitudinal organisation "
-                        "from BIDS specification. Clinica does not know how to handle cross sectional dataset, but it "
-                        "can convert it to a Clinica compliant form (using session ses-M00)"
-                    ),
-                    lvl="warning",
-                )
+            try:
+                detect_cross(all_subs, bids_dir, cross_subj, long_subj)
+            except ClinicaInconsistentDatasetError:
```

Comment:
Replace with:

```python
cross_subj, long_subj = whatever_name_you_chose(all_subs, bids_dir)
try:
    _check_cross_subj(cross_subj)
except ClinicaInconsistentDatasetError as e:
    cprint(e, lvl="warning")
```

The check function would look like:

```python
def _check_cross_long_subj(cross_subj: list) -> None:
    if len(cross_subj) > 0:
        raise ClinicaInconsistentDatasetError(cross_subj)
```

and unindent the rest